app_dart/lib/src/request_handlers/github_rate_limit_status.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+import '../datastore/cocoon_config.dart';
+import '../foundation/utils.dart';
+import '../request_handling/body.dart';
+import '../request_handling/request_handler.dart';
+import '../service/github_service.dart';
+
+/// Endpoint to collect the current GitHub API quota usage of Cocoon on all supported Flutter repos.
```

Comment:
Can we add documentation on the format of the data being collected? I'm particularly interested on how are we managing overlapping windows.

Reply:
Added documentation on the BQ entry format. Overlapping windows is managed by locking this behind a CacheRequestHandler to ensure its only run once a minute.

In general, BQ documentation in Cocoon is lacking. I created https://github.com/flutter/flutter/issues/75959 to track.

Reply:
Do we still need to align the windows with the server? usually the APIs provide the necessary information in the form of headers.

Reply:
There's an explicit rate limit api call that's being used here. I opted for this as it was simpler and didn't see a need for monitoring more fine grained than a minute.